### DISCUSSION TITLE
zxz - a CLI for quick file sharing via https://0x0.st

### DISCUSSION THREAD

**Discussion by meleu** [bashly-framework/bashly#599] [Show and tell]
https://github.com/meleu/zxz

I've created this mainly to try bashly. Turns out that the resulting program is actually useful. :)

It basically allows you to:

- upload files to 0x0.st
- list the files you uploaded
- delete the files you uploaded

Again, I'd like to thank @DannyBen for creating such a useful tool! :heart: 

Thanks to bashly, it's now so smooth to add/change/remove things like:
- a new `--option` (with or without args)
- a new subcommand
- help messages

Just like shellcheck, shfmt and bats, this seems to be the kind of tool that will change forever the way I write my bash programs.

**Comment by DannyBen** [comment]
That is cool. First I've heard of 0x0.st - seems like a nice tool to know.
I am adding this to [the list](https://bashly.dannyb.co/examples/#real-world-examples).

**Comment by DannyBen** [comment]
If you create a GitHub release, and put the generated script as an attached file to the release, people can install it with just

```bash
eget meleu/zxz
# or 
eget --upgrade-only --to=/usr/local/bin meleu/zxz
```

**Reply by meleu** [reply]
I wasn't aware about this [eget tool](https://github.com/zyedidia/eget).

**Comment by DannyBen** [comment]
Another suggestion:

Move the `dependencies` to the first command that needs it, mark it with a YAML anchor and reuse it in subsequent commands.

```yaml
# inside the first command:
dependencies: &deps
  - curl
  - gum
  - cut
  - date
  - grep

# in other commands:
dependencies: *deps
```

This way, people can still run `zxz help` without having the dependencies.

Furthermore - if you do it this way, you gain the ability to fine tune the dependencies on a per command basis and only require what is actually being used.
